### ISSUE TITLE
Преимущества использования Gemini против ImageMagic

### ISSUE THREAD

**Issue by jhenya** [bem/bem-forum-content-ru#745]
Всем привет
Коллеги, у меня концептульный вопрос. 
Мы на проекте используем Behat + PHPUnit и раним все это дело через Bamboo на виртуалках. Также настрена многопоточность выполнения features через parallel  и объединение всех резалтов в один консолидированный репорт. 
На сервер Бамбу был установлен ImageMagic он входит в стандартные пакеты дистриюутива RH. Доступ к API ImageMagic  осуществляется через нативные методы PHP класса Imagick
Таким образом имея уже все описаные элементы страниц в функциональных тестах лекго их переиспользовать для визуальногт тестирования, что я и сделал. Get элемент на странице, передаю его в свой метод assertImage, который делает скрин страницы вырезает по координатам нужный объект и кладет его в папку с эталонами либо сравнивает с подготовленным эталоном в зависимости от конфига.  
Результаты сравнения (etalon, actual, diff) выводятся   в тот же репорт что и функциональные тесты таким образом мы используя наш фреймворк покрываем функциональное и визуальное тестирование и получаем один отчет.
Результаты такого визаульного тестирования стабильные.

Использование Gemini подразумеват как минимум дублирование селекторов в другой туле, другой язык тестов, другой очет и много другого. Также  наш фреймворк работает со всеми браузерами и лекго скролит страницу к нужному элементу и делает скрин нужного элемента. 

Есть ли смысл тогда выносить визуальное тестирвание в отдельныую тулу Gemini ? 
Может есть какие-то очевидные минусы нашего подхода, которые мне на первый взгляд не видны и какие-то явные преимущества, которые должны заставить переписать все элементы в другой туле и сапортить ее в дальнейшем
Спасибо

**Comment by tadatuta** [comment]
@sipayRT @arikon

**Comment by tadatuta** [comment]
Если все работает и устраивает, то переезжать смысла нет.

**Comment by jhenya** [comment]
насколько я понимаю сама операция сравнения картинок делается по одному и тому же принципу разница в написании самих тестов и генерации отчетов

**Comment by arikon** [comment]
> насколько я понимаю сама операция сравнения картинок делается по одному и тому же принципу, разница в написании самих тестов и генерации отчетов

@jhenya Нет. В gemini используется более умное сравнение. За подробностями смотри модуль [looks-same](https://github.com/gemini-testing/looks-same).

**Comment by jhenya** [comment]
подскажи плиз в чем именно заключается более умное сравнение?
